scalafix-rules/src/main/scala-3/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -0,0 +1,66 @@
+package scalafix.internal.rule
+
+import scala.meta.*
+
+import metaconfig.Configured
+import scalafix.internal.pc.ExplicitResultTypesFallback
+import scalafix.patch.Patch
+import scalafix.v1.*
+
+
+final class ExplicitResultTypes(
```

Comment:
As I understand it from a very quick review as I am time-constrained but I see you progress very well, the 2.13 version is able to support 3 sources via the dynamic resolution. So we might not need to publish scalafix-rules_3 at all if we stick to that strategy (pending my other comment)?

Reply:
I guess that's what you meant in your previous comment

> Also I commented out using 2.13 compiled rules for Scala 3, but I can make it work with coursier.

Reply:
That would allow to push this faster as https://github.com/scalacenter/scalafix/issues/1680 wouldn't be blocking.

Reply:
>  So we might not need to publish scalafix-rules_3 at all if we stick to that strategy (pending my other comment)?

Yep, it wouldn't be needed, but dynamic resolution in this case is a must.

Reply:
I would favor static resolution over ability to run ExplicitTypes with scalafix-rules_2.13. I can dedicate my full Tuesday next week to https://github.com/scalacenter/scalafix/issues/1680.

Reply:
Should I wait with the changes here until your PR is finished then?

Reply:
I think both PR are orthogonal, it's just that your implementation will only be available to users after my upcoming PR. So you can iterate now on this one if you have time, and we can merge it as soon as it's ready independently of mine.